app/controllers/publish/courses/age_range_controller.rb

```diff
@@ -87,12 +85,24 @@ def current_step
         :age_range
       end
 
+      def build_provider
+        return if params[:provider_code].blank?
+
+        @provider = @recruitment_cycle.providers.find_by!(
+          provider_code: params[:provider_code].upcase,
+        )
+      end
+
+      def build_recruitment_cycle
+        @recruitment_cycle = RecruitmentCycle.find_by(
+          year: params[:recruitment_cycle_year],
+        ) || RecruitmentCycle.current_recruitment_cycle
+      end
+
       def build_course
```

Comment:
Can be dropped and as it should be using

https://github.com/DFE-Digital/teacher-training-api/blob/db1ca9b0b3a8c8d999e544cf1028766f1814b47b/app/models/concerns/publish/course_basic_detail_concern.rb#L60-L62